feat(BingoBoard, BingoItem) : 빙고 수정 기능

holeman79 opened the pull request:
- 빙고 수정 기능
- ktlint 제거
- 빙고 수정 api 문서

holeman79 committed:
feat(BingoBoard, BingoItem) : 빙고 수정 기능

- 빙고 수정 기능
- ktlint 제거
- 빙고 수정 api 문서

holeman79 committed:
feat(BingoPeriod) : base update에 since, until 추가

seongheonsim commented on src/main/kotlin/com/beside/groubing/groubingserver/domain/bingo/api/BingoItemUpdateApi.kt:
```diff
@@ -0,0 +1,31 @@
+class BingoItemUpdateApi(
+    private val bingoItemUpdateService: BingoItemUpdateService
+) {
+    @PutMapping("/{id}/bingo-items/{bingoItemId}")
```
@holeman79 
원진님, 빙고 보드 관련된 API 의 엔드포인트가 `/bingoboards` 로 되어 있어서 빙고 아이템도 통일하면 좋을 것 같다는 생각이 드는데 원진님 의견은 어떠신지 궁금합니다! (개인적으로는 하이픈을 사용해서 통일하는 것이 어떨까 싶습니다..ㅎㅎ)

- 하이픈 미사용
  - `/bingoboards`
  - `/bingoitems`
- 하이픈 사용
  - `/bingo-boards`
  - `/bingo-items`

holeman79 replied:
음 그렇네요ㅎㅎ rest url은 하이픈 사용하는 쪽으로 통일하는게 어떨까요? 보통 하이픈은 많이 사용해서요!

seongheonsim commented on src/main/kotlin/com/beside/groubing/groubingserver/domain/bingo/payload/request/BingoBoardBaseUpdateRequest.kt:
```diff
@@ -0,0 +1,26 @@
+import org.hibernate.validator.constraints.Length
+
+class BingoBoardBaseUpdateRequest(
+    @NotBlank(message = "제목을 입력해 주세요.")
```
@holeman79 
코틀린에서는 Validation 애노테이션을 활성화 하려면 별도로 다음과 같이 타겟을 설정해줘야 유효성 검사를 수행하더라구요.

```kotlin
@field:NotBlank(message = "제목을 입력해 주세요.")
```

[해당 문서](https://www.baeldung.com/kotlin/valid-spring-annotation) 참고 해주시면 될 것 같습니다!

holeman79 replied:
그러네요.... 수정해서 다시 올려놓도록 하겠습니다ㅎㅎ

seongheonsim commented on src/test/kotlin/com/beside/groubing/groubingserver/domain/bingo/api/BingoBoardUpdateApiTest.kt:
```diff
@@ -0,0 +1,205 @@
+    }
+})
+
+private fun checkUpdateResponse(
```
@holeman79 
해당 함수는 MockMvc 클래스의 확장 함수로 만들면 여러 테스트 코드에서도 활용 가능할 것 같아요!

```kotlin
fun MockMvc.checkUpdateResponse(
  url: String,
  path: Collection<String>? = null,
  request: String,
  response: String,
  documentationIdentifier: String,
  snippet: Collection<Snippet>? = null
) {
  ...
}
```

seongheonsim commented on src/main/kotlin/com/beside/groubing/groubingserver/domain/bingo/api/BingoBoardUpdateApi.kt:
```diff
@@ -0,0 +1,70 @@
+        return ApiResponse.OK(bingoBoardResponse)
+    }
+
+    @PatchMapping("/{id}/members-period")
```
@holeman79 
원진님, 혹시 해당 API 는 그룹 빙고 발행 시 호출해야 하는 API 인가요~?

holeman79 replied:
아뇨 이건 빙고보드 제목이랑 목표 빙고, 기간 수정할 때 호출하는 API입니다. 빙고보드 기본 정보를 변경한다 생각해서 이름은 base로 지정했습니다. 와이어프레임 US-06번 부분입니다.

seongheonsim replied:
@holeman79 
앗, 저는 `/bingoboards/{id}/member-period` 에 대한 내용인데, 제가 리뷰를 엄한 곳에 했나봐요. 죄송합니다..🥲
처음에 `/members-period` 라는 엔드포인트를 봤을 때 유저의 기간을 수정하는건가 라고 생각해서 해당 API 의 비즈니스 로직을 확인 해보니까 빙고 멤버를 추가하는 로직이 있더라구요. 그래서 그룹 빙고 발행할 때 호출하는 API 라면 다른 방식으로 엔드포인트를 표현하면 어떨까 싶어서 리뷰 남겼습니다!ㅎㅎ

holeman79 replied:
아 이건 마지막 발행할 때 빙고 멤버와 기간 입력하는 API인데 음 헷갈릴만 하군요ㅎㅎ 두개를 합친 명칭이 딱 떠오르지 않아서..... 발행이라는 이름으로 바꾸는게 나을 수도 있겠네요ㅎㅎ publish나 이런 명칭으로 변경할게요!

seongheonsim reviewed (changes requested):
@holeman79 
안녕하세요, 원진님! 
올려주신 PR 확인 후 몇 가지 리뷰 남겼으니 한 번 부탁 드립니다.
고생 많으셨습니다!🫡

holeman79 committed:
feat(BingoBoard) : 성헌님 리뷰사항 받아서 다시 수정

- api url : bingoboards -> bingo-boards
- @field: API 입력받는 request Dto validation 적용될 수 있게 앞에 @field: 적용
- members-period -> publish-info 로 url 변경

seongheonsim reviewed (approved):
@holeman79 
원진님, 수정해주신 부분 확인 했습니다. 해당 부분 머지하도록 하겠습니다! 고생하셨어요..!👍
